File: qsort/qsort.py

def qsort(lst):
	if len(lst) < 2:
		return lst
	pivot = lst[0]
	return qsort([x for x in lst if x < pivot]) + \
		   		 [x for x in lst if x == pivot] + \
		   qsort([x for x in lst if x > pivot])

print(qsort([1,3,2,4,5]))

File: qsort/qsort.hs

qsort lst = 
    if length lst < 2 then lst
    else
        let pivot = lst !! 0
        in qsort([x | x <- lst, x < pivot]) ++ 
                 [x | x <- lst, x == pivot] ++ 
           qsort([x | x <- lst, x > pivot])

main = print (qsort [2,5,3,4,1])
